server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionFinalizer.java

```diff
@@ -155,7 +155,7 @@ private void processPending() {
 
         Map<KeyExtent,TabletMetadata> tabletsMetadata;
         var extents = batch.stream().map(ExternalCompactionFinalState::getExtent).collect(toList());
-        try (TabletsMetadata tablets = context.getAmple().readTablets().forTablets(extents)
+        try (TabletsMetadata tablets = context.getAmple().readTablets().forTablets(extents, e -> {})
```

Comment:
What's going on with this NOOP consumer?

Reply:
Thats for when the operation is ok with only getting a subset of what it asked for.

Reply:
I made this comment before making my other comment. I think in the absence of a named lambda, a small inline comment explaining `e -> {}` would help. At a glance, it looks like a bug, because it looks like you're iterating over tablets, but not doing anything with them (because `forTablets` looks like a terminal operation like `forEach`), so this looks like `forEach(e -> { /* noop */})`. But naming the lambda as I suggested in my other comment is the most concise explanation that the NOOP consumer was chosen on purpose.

Reply:
Since there can be a performance advantage when the consumer is not needed, I think optional makes that intention very clear and the code can avoid trying to compute the missing extents.